app/models/ability.rb

```diff
@@ -46,6 +46,11 @@ def initialize(user)
     end
 
     # supervisor
+    if user.supervisor?
+      can :read, :all, User do |other_user|
```

Comment:
It looks like this is mixing up things a bit 🤔  From how I read the issue, we only need a new ability `read_email` here, to allow the supervisor to access a user's email address. The object passed it will be a user as well, not just an email address.